cps2pc

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python
+
+import argparse
+import cps
+import os
+import re
+import subprocess
+import sys
+
+from collections import OrderedDict
+
+_default_prefix = '${pcfiledir}/../..'
+
+_supported_compilers = ['gcc', 'clang']
+_supported_kernels = ['cygwin', 'linux', 'darwin']
+_supported_cpp_runtime_vendors = {'gnu': 'stdc++', 'llvm': 'c++'}
+
+# Compile definitions
+_re_definitions = re.compile('(!)?(.*)')
+_unix_definition_prefixes = {None: '-D', '!': '-U'}
+_compilers_definition_prefixes = {
+    'gcc': _unix_definition_prefixes, 'clang': _unix_definition_prefixes}
+
+# Compile features
+_re_we_features = re.compile('((no)?(warn|error):(error$)?)(.*)')
+_W_we_prefixes = {
+    'warn:error': '-Werror', 'nowarn:error': '-Wno-error', 'warn:': '-W',
+    'nowarn:': '-Wno-', 'error:': '-Werror=', 'noerror:': '-Wno-error='}
+_compilers_we_prefixes = {'gcc': _W_we_prefixes, 'clang': _W_we_prefixes}
+_supported_compiler_feature_list = [
+    'c99', 'c11', 'c++03', 'c++11', 'c++14', 'c++17', 'gnu']
+_supported_gcc_features = {
+    x: '--std=%s' % x for x in _supported_compiler_feature_list}
+_supported_clang_features = _supported_gcc_features
+_supported_compile_features = {
+    'gcc': _supported_gcc_features, 'clang': _supported_clang_features}
+
+# Link features
+_supported_link_features = {x: {'threads': '-lpthread'}
+                            for x in _supported_kernels}
+
+#------------------------------------------------------------------------------
+def get_link_features(link_features, platform):
+    try:
+        features = []
+        for feature in link_features:
+            features.append(_supported_link_features[platform][feature])
+        return features
+    except KeyError as k:
+        raise Exception('Unknown link feature %s on %s' %
```

Comment:
Failing here might be overkill... do you think we really need to make this a fatal error, or should we just warn about it and slog on?